### PR TITLE
Fix documentation warnings

### LINKED ISSUE #2699: Documentation warnings deserving attention
While further tweaking the documentation generation the other day, I notice some warnings go past that might be worth our attention. They include:

```
INFO     -  The following pages exist in the docs directory, but are not included in the "nav" configuration:
              - api/content_switcher.md
              - api/validation.md
              - examples/styles/README.md
WARNING  -  griffe: textual/app.py:881: No type or annotation for parameter '*args'
WARNING  -  griffe: textual/app.py:882: No type or annotation for parameter '**kwargs'
WARNING  -  griffe: textual/dom.py:1077: No type or annotation for parameter '**update_styles'
WARNING  -  Documentation file 'guide/workers.md' contains a link to 'guide/textual.worker.Worker' which is not found in the documentation files.
WARNING  -  mkdocs_autorefs.plugin: widgets/input.md: Could not find cross-reference target '[textual.suggester.Suggester]'
```

### PR DESCRIPTION
**Pull request by guy-av**
**Please review the following checklist.**

- [x] Updated documentation
- [x] For some reason `textual.app.App.compose()` doesn't seem to appear in the reference section of the docs hence breaking a specific link on the `App basics` page.

Resolves #2699

### CODE REVIEW

**Comment by rodrigogiraoserrao** [comment]
The issue with the compose method is that it is excluded from the docs, but if it is linked from elsewhere, we either link to something else or we make sure that App.compose is documented as well.
Look at the api documentation for Widget, where we override some methods that are otherwise excluded via the documentation configuration.

**Comment by guy-av** [comment]
@davep made sure the filters in `api/app.md` match those of the widgets, filtering out only `__init__` and private members. Will you take another look please?

File: mkdocs-nav.yml [hunk 1]
```diff
@@ -165,6 +165,7 @@ nav:
           - "widgets/tabbed_content.md"
           - "widgets/tabs.md"
           - "widgets/text_area.md"
+          - "widgets/toast.md"
```
Comment (davep) [inline mkdocs-nav.yml:168]: My memory is hazy, but I'm sure we left toast out of the widget list as it's not really a "public-facing" widget; it's more of an internal support widget for the default implementation of `App.notify`.

File: mkdocs-nav.yml [hunk 2]
```diff
@@ -174,6 +175,7 @@ nav:
       - "api/color.md"
       - "api/command.md"
       - "api/containers.md"
+      - "api/content_switcher.md"
```
Comment (davep) [inline mkdocs-nav.yml:178]: Given that `ContentSwitcher` is within the widgets, I'm wondering why there is an `api/content_switcher.md` at all.

**Comment by davep** [comment]
> made sure the filters in api/app.md match those of the widgets, filtering out only __init__ and private members. Will you take another look please?

@guy-av Forgive me, I may have missed some context here. Another look? Is this something I've looked at before?

**Comment by guy-av** [comment]
> > made sure the filters in api/app.md match those of the widgets, filtering out only **init** and private members. Will you take another look please?
> 
> @guy-av Forgive me, I may have missed some context here. Another look? Is this something I've looked at before?

Actually didn't mean to tag you 😅 sorry about that.
Thanks for reviewing though!

File: src/textual/widgets/_option_list.py [hunk 3]
```diff
@@ -825,7 +825,7 @@ def get_option(self, option_id: str) -> Option:
         """
         return self.get_option_at_index(self.get_option_index(option_id))
 
-    def get_option_index(self, option_id):
+    def get_option_index(self, option_id: str):
```
Comment (willmcgugan) [inline src/textual/widgets/_option_list.py:828]: Can we have a return type here.

**Review by willmcgugan — CHANGES_REQUESTED** [review]

**Comment by rodrigogiraoserrao** [comment]
I'm closing this as it got stale and was superseded by the PR linked above.

**Pull request closed without merging**